spec/syntax.md

```diff
@@ -669,6 +651,46 @@ reserved-char = %x00-08        ; omit HTAB and LF
               / %xE000-10FFFF
 ```
 
+## Markup
+
+**_<dfn>Markup</dfn>_** _placeholders_ are _pattern_ parts
+that can be used to represent non-language parts of a _message_,
+such as inline elements or styling that should apply to a span of parts.
+Markup comes in three forms:
+
+**_<dfn>Markup-open</dfn>_** starts with U+0023 NUMBER SIGN `#` and
+represents an opening element within the _message_,
+such as markup used to start a span.
+It MAY include _options_.
+
+**_<dfn>Markup-standalone</dfn>_** starts with U+0023 NUMBER SIGN `#`
+and has a U+002F SOLIDUS `/` immediately before its closing `}`
+representing a self-closing or standalone element within the _message_.
+It MAY include _options_.
+
+**_<dfn>Markup-close</dfn>_** starts with U+002F SOLIDUS `/` and
+is a _pattern_ part ending a span.
+Unlike the other forms, it does not include _options_.
+
+```abnf
+markup       = "{" [s] markup-open [s] ["/"] "}"
+             / "{" [s] markup-close [s] "}"
+markup-open  = "#" identifier *(s option)
+markup-close = "/" identifier
+```
+
+> A _message_ with one `button` markup span and a standalone `img` markup element.
+>
+> ```
+> {#button}Submit{/button} or {#img alt=|Cancel| /}.}
+> ```
+
+_Markup-open_ and _markup-close_ parts MAY be paired,
+but this is not a requirement.
+A _markup-open_ MAY be present in a message without
+a corresponding _markup-close_, and vice versa.
+_Markup-standalone_ parts are not paired.
```

Comment:
@macchiati @aphillips Would this work?

```suggestion
No validity requirements are imposed for the proper pairing or nesting of _markup_.
A _markup-open_ MAY be present in a message without
a corresponding _markup-close_, and vice versa.
```

Reply:
I noodled with this for a while. I think we want to allow markup regimes to impose requirements while saying that the message is otherwise "valid".

```suggestion
A _markup-open_ can appear without a corresponding _markup-close_.
A _markup-close_ can appear without a corresponding _markup-open_.
_Markup_ _placeholders_ can appear in any order without making the _message_ invalid.
However, specifications or implementations defining _markup_ might impose requirements
on the pairing, ordering, or contents of _markup_ during _formatting_.
```